app/javascript/src/components/Dashboard/Notes/NewNoteForm.jsx

```diff
@@ -1,49 +1,110 @@
+      dueDate: values.dueDate
+        ? dayjs(values.dueDate, "DD/MM/YYYY").format("MMM DD, YYYY")
+        : "",
+      createdDate: dayjs(new Date(), "DD/MM/YYYY").format("MMM DD, YYYY"),
+      tags: values.tags.charAt(0).toUpperCase() + values.tags.substring(1)
+    };
     try {
-      await notesApi.create(values);
-      refetch();
+      setSubmitted(true);
+      noteDispatch({
+        type: "ADD_NOTE",
+        payload: { note }
+      });
       onClose();
+      Toastr.success("Note added successfully");
     } catch (err) {
       logger.error(err);
+    } finally {
+      setSubmitted(false);
     }
   };
+
+  const getOptions = options => {
+    return options.map(opt => ({ label: opt, value: opt.toLowerCase() }));
+  };
+
   return (
     <Formik
-      initialValues={{
-        title: "",
-        description: ""
-      }}
+      initialValues={formInitialValues.newNoteform}
+      validateOnBlur={submitted}
+      validateOnChange={submitted}
       onSubmit={handleSubmit}
-      validationSchema={yup.object({
-        title: yup.string().required("Title is required"),
-        description: yup.string().required("Description is required")
-      })}
+      validationSchema={formValidationSchemas.newNoteform}
     >
-      {({ isSubmitting }) => (
+      {({ isSubmitting, values, setFieldValue }) => (
         <Form>
-          <Input label="Title" name="title" className="mb-6" />
-          <Textarea label="Description" name="description" rows={8} />
+          <Input label="Note Title" name="title" className="mb-6" />
+          <Select
+            placeholder="Select an option"
+            label="Tags"
+            className="mb-6"
+            isDisabled={false}
+            name="tags"
+            onChange={tag => setFieldValue("tags", tag.value)}
+            options={getOptions(constants.TAGS)}
+          />
+          <Textarea
+            label="Note Description"
+            name="description"
+            rows={8}
+            className="mb-6"
+          />
+          <Select
+            placeholder="Select an option"
+            label="Assigned Contact"
+            className="mb-6"
+            isDisabled={false}
+            name="contact"
+            onChange={assignee => setFieldValue("contact", assignee.value)}
+            options={getOptions(constants.CONTACTS)}
+          />
+          <Switch
+            className="mb-6"
+            label="Add Due Date to Note"
+            onChange={() => setSwitchCheckbox(!checked)}
+            checked={checked}
+          />
+          {checked && (
+            <DateInput
+              selected={values.dueDate}
+              format="DD/MM/YYYY"
+              minDate={new Date()}
+              value={formInitialValues.newNoteform.due_date}
+              label="Due Date"
+              onChange={date => setFieldValue("dueDate", date)}
+            />
+          )}
           <div className="nui-pane__footer nui-pane__footer--absolute">
```

Comment:
Rather than using `mb-6` on each individual element, we can use `space-y-6`. 

Please refer [space between](https://tailwindcss.com/docs/space#class-reference) from tailwindCSS for more details.